chain/chain/src/flat_storage_creator.rs

```diff
@@ -0,0 +1,114 @@
+use crate::{ChainStore, ChainStoreAccess, RuntimeAdapter};
+use crossbeam_channel::{unbounded, Receiver, Sender};
+use near_chain_primitives::Error;
+use near_primitives::types::{BlockHeight, ShardId};
+use near_store::flat_state::FlatStorageStateStatus;
+use std::sync::{Arc, Mutex};
+use tracing::info;
+
+/// Number of parts to which we divide shard state for parallel traversal.
+// TODO: consider changing it for different shards, ensure that shard memory usage / `NUM_PARTS` < X MiB.
+#[allow(unused)]
+const NUM_PARTS: u64 = 4_000;
+
+/// Number of traversed parts during a single step of fetching state.
+#[allow(unused)]
+const PART_STEP: u64 = 50;
+
+/// Creates flat storage and tracks creation status for the given shard.
```

Comment:
Comments in this module could benefit from adding more context. It's tautological that flat storage creator creates flat storage. 

What isn't obvious from the code is why&when&how we'd do that. 

Something like "we want to create flat storage for X, but the amount of data is big, so we do this in parallel, in background" (As I am not following the work closely, I actually don't know the flow here, so can't suggest exact wording :) )

Reply:
Oops, got it auto-merged :|

Reply:
I'll create another PR with actual creation code, add more context there and tag you